### DISCUSSION TITLE
my backend is getting error

### DISCUSSION THREAD

**Discussion by thejastm15** [vercel/community#7487] [Help]
### Question

i am getting this error
1) [GET] /api/users/me status=404

2)[Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
  digest: '4012416115'
}

3)SyntaxError: Unexpected token 'I', "Infinite l"... is not valid JSON
    at JSON.parse (<anonymous>)
    at parseJSONFromBytes (node:internal/deps/undici/undici:4306:19)
    at successSteps (node:internal/deps/undici/undici:4288:27)
    at fullyReadBody (node:internal/deps/undici/undici:2724:9)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async consumeBody (node:internal/deps/undici/undici:4297:7)
    at async getServerSideUser (/var/task/.next/server/chunks/611.js:1:33569)
    at async Navbar (/var/task/.next/server/chunks/611.js:1:26858)

**Comment by Dhanraj-Awatade** [comment]
I'm having the same issue. Please tell if it got resolved.

**Comment by amyegan** [comment]
It looks like something other than JSON data was returned. You could try wrapping some of the code in a [try...catch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) to log more details and handle the error.

**Reply by thejastm15** [reply]
I tried but it is not working

**Reply by amyegan** [reply]
Can you show me what you mean? It would help if I knew what you're seeing

**Reply by Dhanraj-Awatade** [reply]
I actually fixed this by wrapping the fetch function inside try catch as @amyegan already suggested. The root cause though was Nextjs returning a webpage (404 I guess) because of the CORS error.

P.S: I still haven't been able to resolve CORS preflight redirect error on vercel but I tried on railway.app and setting the NEXT_PUBLIC_SERVER_URL env variable to the temporary domain provided by railway fixed the CORS error.

[a comment by an automated account is not shown]
